projects/igniteui-angular/src/lib/directives/input/input.directive.spec.ts

```diff
@@ -917,6 +1013,63 @@ class ToggleRequiredWithNgModelInputComponent {
     public data1 = '';
     public isRequired = false;
 }
+@Component({
+    template: `
+        <form [formGroup]="reactiveForm" (ngSubmit)="onSubmitReactive()">
+            <igx-input-group #igxInputGroup>
+                <input igxInput #inputReactive name="fullName" type="text" formControlName="fullName" />
+                <label igxLabel for="fullName">Full Name</label>
+                <igx-suffix>
+                    <igx-icon>person</igx-icon>
+                </igx-suffix>
+            </igx-input-group>
+        </form>
+`
+})
+
+class InputReactiveFormComponent {
+    @ViewChild('igxInputGroup', { static: true }) public igxInputGroup: IgxInputGroupComponent;
+    @ViewChild('inputReactive', { read: IgxInputDirective }) public input: IgxInputDirective;
+    public reactiveForm: FormGroup;
+
+    public validationType = {
+        fullName: [Validators.required]
+    };
+
+    constructor(fb: FormBuilder) {
+        this.reactiveForm = fb.group({
+            fullName: new FormControl('', Validators.required)
+        });
+    }
+    public onSubmitReactive() { }
+
+    public removeValidators(form: FormGroup) {
+        // eslint-disable-next-line guard-for-in
+        for (const key in form.controls) {
+            form.get(key).clearValidators();
+            form.get(key).updateValueAndValidity();
+        }
+    }
+
+    public addValidators(form: FormGroup) {
+        // eslint-disable-next-line guard-for-in
+        for (const key in form.controls) {
+            form.get(key).setValidators(this.validationType[key]);
+            form.get(key).updateValueAndValidity();
+        }
+    }
+
+    public markAsTouched() {
+        if (!this.reactiveForm.valid) {
+            for (const key in this.reactiveForm.controls) {
```

Comment:
The same can be applied here as well though the check on the next line seems to make eslint happy as it seems to be explicitly looking for a check inside the `for in` loop.